src/traces/scattergeo/calc.js

```diff
@@ -10,45 +10,34 @@
 'use strict';
 
 var isNumeric = require('fast-isnumeric');
+var BADNUM = require('../../constants/numerical').BADNUM;
 
 var calcMarkerColorscale = require('../scatter/colorscale_calc');
-
+var arraysToCalcdata = require('../scatter/arrays_to_calcdata');
 
 module.exports = function calc(gd, trace) {
-    var hasLocationData = Array.isArray(trace.locations),
-        len = hasLocationData ? trace.locations.length : trace.lon.length;
-
-    var calcTrace = [],
-        cnt = 0;
+    var hasLocationData = Array.isArray(trace.locations);
+    var len = hasLocationData ? trace.locations.length : trace.lon.length;
+    var calcTrace = new Array(len);
 
     for(var i = 0; i < len; i++) {
-        var calcPt = {},
-            skip;
+        var calcPt = calcTrace[i] = {};
 
         if(hasLocationData) {
             var loc = trace.locations[i];
-
-            calcPt.loc = loc;
-            skip = (typeof loc !== 'string');
+            calcPt.loc = typeof loc === 'string' ? loc : null;
         }
         else {
-            var lon = trace.lon[i],
-                lat = trace.lat[i];
+            var lon = trace.lon[i];
+            var lat = trace.lat[i];
+            calcPt.lonlat = new Array(2);
 
-            calcPt.lonlat = [+lon, +lat];
-            skip = (!isNumeric(lon) || !isNumeric(lat));
+            calcPt.lonlat[0] = isNumeric(lon) ? +lon : BADNUM;
+            calcPt.lonlat[1] = isNumeric(lat) ? +lat : BADNUM;
```

Comment:
probably all we'd need to do for https://github.com/plotly/plotly.js/pull/1538/files#r109043439 to work is change this to something like
```js
if(isNumeric(lon) && isNumeric(lat)) calcPt.lonlat = [+lon, +lat];
else calcPt.lonlat = [BADNUM, BADNUM]; // or even [BADNUM] or []
```